aws/resource_aws_cloudwatch_event_archive.go

```diff
@@ -0,0 +1,200 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	events "github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsCloudWatchEventArchive() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCloudWatchEventArchiveCreate,
+		Read:   resourceAwsCloudWatchEventArchiveRead,
+		Update: resourceAwsCloudWatchEventArchiveUpdate,
+		Delete: resourceAwsCloudWatchEventArchiveDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCloudWatchEventArchiveName,
+			},
+			"event_source_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 512),
+			},
+			"event_pattern": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateEventPatternValue(),
+				StateFunc: func(v interface{}) string {
+					json, _ := structure.NormalizeJsonString(v.(string))
+					return json
+				},
+			},
+			"retention_days": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCloudWatchEventArchiveCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatcheventsconn
+
+	input, err := buildCreateArchiveInputStruct(d)
+
+	if err != nil {
+		return fmt.Errorf("Creating CloudWatch Events Archive parameters failed: %w", err)
+	}
+
+	log.Printf("[DEBUG] Creating CloudWatch Events Archive: %s", input)
+
+	_, err = conn.CreateArchive(input)
+	if err != nil {
+		return fmt.Errorf("Creating CloudWatch Events Archive failed: %w", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+
+	log.Printf("[INFO] CloudWatch Events Archive (%s) created", d.Id())
+
+	return resourceAwsCloudWatchEventArchiveRead(d, meta)
+}
+
+func resourceAwsCloudWatchEventArchiveRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).cloudwatcheventsconn
+	input := &events.DescribeArchiveInput{
+		ArchiveName: aws.String(d.Id()),
+	}
+
+	out, err := conn.DescribeArchive(input)
+
+	if err != nil {
+		return fmt.Errorf("Error reading CloudWatch Events archive: %w", err)
+	}
+
+	log.Printf("[DEBUG] Found Archive: #{*out}")
+
+	// Review Question - Is there a problem in setting more than should name and Arn?
```

Comment:
I don't see a problem, but to make sure I fully understand, what is your concern?

Reply:
Thanks @bill-rich - As it's my first time, I wasn't sure if there was a limit on how many you could set, that's all :)